src/NuGet.Clients/NuGet.VisualStudio.Common/Services/NuGetFeatureFlagService.cs

```diff
@@ -45,7 +45,7 @@ public async Task<bool> IsFeatureEnabledAsync(NuGetFeatureFlagConstants featureF
                 var featureFlagService = await _ivsFeatureFlags.GetValueAsync();
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                 featureEnabled = featureFlagService.IsFeatureEnabled(featureFlag.Name, defaultValue: featureFlag.DefaultState);
-                _featureFlagCache.Add(featureFlag.Name, featureEnabled);
+                _featureFlagCache[featureFlag.Name] = featureEnabled;
```

Comment:
Should we use [`ConcurrentDictionary`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-8.0) instead of `Dictionary` to avoid concurrency issues?

Reply:
My guess is the discussion wasn't about whether this can happen - because clearly it does from your feedback ticket - but rather, it won't matter if multiple calls result in updating the cache as it should always be the same value (the service won't read settings multiple times). Is that right?

Instead of allowing the cache to update again, what about another checking `if` the key exists immediately after switching to the main thread? No reason to call the `featureFlagService` again if the reasoning I gave above is right.

Reply:
So adding another TryGetValue? 

            if (!_featureFlagCache.TryGetValue(featureFlag.Name, out bool featureEnabled))
            {
                var featureFlagService = await _ivsFeatureFlags.GetValueAsync();
                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                if (!_featureFlagCache.TryGetValue(featureFlag.Name, out featureEnabled))
                {
                    featureEnabled = featureFlagService.IsFeatureEnabled(featureFlag.Name, defaultValue: featureFlag.DefaultState);
                    _featureFlagCache[featureFlag.Name] = featureEnabled;
                }
            }

Reply:
Yea that's correct, which is part of what we discussed during the PR call today, there shouldn't be two items updating the cache at the same time. I think this is a decent compromise though. We could also just do TryAdd 🤷